app/api/sponsors.py

```diff
@@ -109,7 +111,8 @@ class SponsorDetail(ResourceDetail):
     """
     Sponsor detail by id
     """
-    decorators = (jwt_required,)
+    decorators = (api.has_permission('is_coorganizer', methods="PATCH,DELETE",
```

Comment:
add, fetch="event_id", fetch_as="event_id", model=Sponsor

This will use <id> of sponsor to fetch event_id and put it into kwargs as "event_id"